### PR TITLE
When client.request failed, throw the exception

### PR DESCRIPTION
**Pull request by rxf113**
fix bug #15

### CODE REVIEW

**Comment by tsegismont** [comment]
cc @vietj

**Comment by vietj** [comment]
Can you add a test ?

**Comment by rxf113** [comment]
> Can you add a test ?

I added a test, and There are some other changes, please confirm carefully

**Comment by rxf113** [comment]
> > Can you add a test ?
> 
> I added a test, and There are some other changes, please confirm carefully

Added a test sample  [vertx-grpc-server-unavailable](https://github.com/rxf113/vertx-grpc-server-unavailable)

**Comment by vietj** [comment]
the test has been removed, can you elaborate ?

**Comment by rxf113** [comment]
> the test has been removed, can you elaborate ?
I added a test sample  [vertx-grpc-server-unavailable](https://github.com/rxf113/vertx-grpc-server-unavailable)

The previous test shouldn't be there because it needs to depend on VetxStub and I had to modify the pom.xml, which I don't think is a qualifying unit test, so i removed it

**Comment by rxf113** [comment]
Updated description：  https://github.com/rxf113/vertx-grpc-server-unavailable/blob/main/README.md

**Review by liangyuanpeng — COMMENTED** [review]
Maybe add test case on this repo would be great.

File: vertx-grpc-client/src/main/java/io/vertx/grpc/client/VertxClientCall.java [hunk 1]
```diff
@@ -121,6 +123,12 @@ public void start(Listener<ResponseT> responseListener, Metadata headers) {
           }
         });
         writeAdapter.init(request, new BridgeMessageEncoder<>(methodDescriptor.getRequestMarshaller(), compressor));
+      } else {
+        //Maybe the client.request will fail, eg: server interrupt
+        Throwable err = ar1.cause();
+        if (err instanceof ConnectException || err instanceof ClosedChannelException){
```
Comment (vietj) [inline vertx-grpc-client/src/main/java/io/vertx/grpc/client/VertxClientCall.java:129]: I think we already have error handling in the second asynchronous result that performs a close operation, we should try merging the two in a `compose` operation and handle a single doClose in the composed future.

**Comment by vietj** [comment]
thanks, I ported the code here in the more general PR about vertx client call error handling : https://github.com/eclipse-vertx/vertx-grpc/pull/19

**Pull request closed without merging**